src/composables/accountSelector.ts

```diff
@@ -0,0 +1,200 @@
+        default:
+          return [];
+      }
+    },
+  );
+  const accountsFiltered = computed(() => {
+    const entries: IAccountSelectorEntry[] = accountsFilteredByType.value ?? [];
+    const searchQueryLower = searchQuery.value.toLowerCase();
+
+    return entries.filter( // Filter by searchQuery
+      ({ name, address }) => [name, address].some(
+        (val) => val.toLowerCase().includes(searchQueryLower),
+      ),
+    ).filter( // Remove duplicates
+      (entry, index, self) => self.findIndex((e) => e.address === entry.address) === index,
+    );
+  });
+
+  function setAccountSelectType(type: AccountSelectTypeFilter, resetProtocolFilter = false) {
+    accountSelectType.value = type;
+    setShowBookmarked(type === ACCOUNT_SELECT_TYPE_FILTER.bookmarked, resetProtocolFilter);
+  }
+  function clearFilters(resetProtocolFilter = false) {
+    accountSelectType.value = ACCOUNT_SELECT_TYPE_FILTER.addressBook;
+    addressBookClearFilters(resetProtocolFilter);
+  }
+
+  watch(
+    protocolFilter,
+    () => {
+      if (protocolFilter.value) {
+        setAccountSelectType(ACCOUNT_SELECT_TYPE_FILTER.addressBook);
+      }
+    },
+  );
+  watch(
+    [accountsTransactionsLatest.value, activeAccount],
+    async () => {
+      latestTransactions.value = await Promise.all(
+        (accountsTransactionsLatest
+          .value[activeAccount.value.address] || []).filter(
+          (transaction: ITransaction) => {
+            const outerTx = transaction.tx!;
+            const innerTx = transaction.tx ? getInnerTransaction(transaction.tx) : null;
+            const { recipientId } = outerTx?.payerId ? outerTx : innerTx;
+            const { direction } = useTransactionData({
+              transaction: toRef(() => transaction),
+            });
+            return direction.value === TX_DIRECTION.sent && recipientId;
+          },
+        ).map(async (transaction: ITransaction): Promise<IAccountSelectorEntry> => {
+          const outerTx = transaction.tx!;
+          const innerTx = transaction.tx ? getInnerTransaction(transaction.tx) : null;
+          const { recipientId } = outerTx?.payerId ? outerTx : innerTx;
+          const middleware = await getMiddleware();
+
+          let name = 'Recipient';
+          const isChainName = recipientId?.startsWith('nm_');
+          const addressBookRecord = getAddressBookEntryByAddress(recipientId);
+
+          if (addressBookRecord) {
+            return {
+              ...addressBookRecord,
+              isAddressBookEntry: true,
+            };
+          }
+          if (isChainName) {
+            name = (await middleware.getName(recipientId)).name;
+          } else {
+            const account = getAccountByAddress(recipientId!);
+            if (account) {
+              name = getName(account.address).value || getDefaultAccountLabel(account);
+            } else {
+              name = getName(recipientId).value || name;
+            }
+          }
```

Comment:
I think the name resolving should not be happen on the logic side, but when rendering the account.
The reason is that in my opinion this composable should not relay on any protocol specific logic.

Reply:
Moved name resolving in `AccountInfo.vue` component, and added a skeleton loader to show it until the name is resolved